include/circt/Dialect/Calyx/CalyxInterfaces.td

```diff
@@ -91,9 +91,15 @@ def CellOpInterface : OpInterface<"CellInterface"> {
       /*methodBody=*/"",
       /*defaultImplementation=*/[{
         SmallVector<circt::calyx::PortInfo> info;
-        for (auto it : llvm::zip($_op->getResults(), $_op.portDirections(),$_op.portNames()))
-          info.push_back(PortInfo{StringAttr::get($_op->getContext(),
-            std::get<2>(it)), std::get<0>(it).getType(), std::get<1>(it)});
+
+        auto zip = llvm::zip($_op->getResults(), $_op.portDirections(), $_op.portNames());
+        for (auto&& [result, direction, name] : zip)
```

Comment:
Is this a C++17 feature? I hadn't seen it before but it's nice.

Reply:
Yep! https://en.cppreference.com/w/cpp/language/structured_binding